src/Symfony/Component/Uid/Ulid.php

```diff
@@ -58,14 +58,31 @@ public static function isValid(string $ulid): bool
      */
     public static function fromString(string $ulid): parent
     {
-        if (36 === \strlen($ulid) && Uuid::isValid($ulid)) {
-            $ulid = (new Uuid($ulid))->toBinary();
-        } elseif (22 === \strlen($ulid) && 22 === strspn($ulid, BinaryUtil::BASE58[''])) {
-            $ulid = BinaryUtil::fromBase($ulid, BinaryUtil::BASE58);
```

Comment:
The implementation was correct and more efficient, I'd prefer to keep it if possible.

Reply:
Do you mean you want to keep all the current code in fromString() and duplicate the logic in the new methods?

Reply:
I've been driven by the implementation here: implementing a selector logic based on caught exceptions is extremely inefficient and not great from a design pov... :)